### PR TITLE
CSharp High-Level API

### PR DESCRIPTION
**Pull request by meyinyil**
This wrapper has been created for using REFPROP.DLL.  In this project, new High-Level API functions are used. Like in excel wrapper, you need only one function for calculating properties. Easier to use and access DLL. 

Functions are given below:

-REFPROPdll()
-SETFLUIDSdll()
-SETPATHdll()

### CODE REVIEW

**Comment by ianhbell** [comment]
First off, please remove all the build files. They should not be part of repo. Same for all project files.

File: wrappers/C#_High_Level_Wrapper/Refprop_Example/Form1.cs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    public partial class Form1 : Form
+    {
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void calculate_Click(object sender, EventArgs e)
+        {
+            Refprop refrpop = new Refprop(); // creating object
```
Comment (ianhbell) [inline wrappers/C#_High_Level_Wrapper/Refprop_Example/Form1.cs:26]: refrpop should be refprop throughout

File: wrappers/C#_High_Level_Wrapper/LICENSE.txt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (ianhbell) [inline wrappers/C#_High_Level_Wrapper/LICENSE.txt:1]: The license should be the top-level license, and by contributing, you agree to use the existing public domain license

File: wrappers/C#_High_Level_Wrapper/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Wrapper of REFPROP for C&#35;
+
+The files in this directory demonstrate how to call REFPROP routines from a C# program for High level API. This example can be used to easily and quickly create your own C# program that uses RefProp.
+Cancel changes
+## License
+
+These files fall under a MIT-style license.
```
Comment (ianhbell) [inline wrappers/C#_High_Level_Wrapper/README.md:7]: See other comment

File: wrappers/C#_High_Level_Wrapper/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+## To Use Solution
+
+1. Do a git clone of the entire repository.
+2. Go into ``wrappers/Csharp`` directory
+3. Open the Visual Studio 2015 or later solution.
+
+## To Use Wraper
+
+1. Donwload the DLL wrapper : [wrappers/C#_High_Level_Wrapper/RefpropDLL/bin/Debug](https://github.com/meyinyil/REFPROP-wrappers/tree/master/wrappers/C%23_High_Level_Wrapper/RefpropDLL/bin/Debug)
```
Comment (ianhbell) [inline wrappers/C#_High_Level_Wrapper/README.md:17]: Should not point to your own repo (dangerous from a security standpoint). Better to provide instructions how to build yourself. Also, default to Release build.

File: wrappers/C#_High_Level_Wrapper/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+2. Insert your project. (Bin/Debug)
```
Comment (ianhbell) [inline wrappers/C#_High_Level_Wrapper/README.md:18]: bin

File: wrappers/C#_High_Level_Wrapper/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+3. Add references the DLL
+4. Create an object.
+5. Here is an example :
+
+             using RefpropDLL;
+             public partial class Refrpop
+             {
+                        Refprop refrpop = new Refprop(); // creating object
```
Comment (ianhbell) [inline wrappers/C#_High_Level_Wrapper/README.md:26]: typo on refrpop

File: wrappers/C#_High_Level_Wrapper/RefpropDLL/Refprop_DLL.cs [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RefpropDLL
+{
+    public class Refprop_DLL
+    {
+        [DllImport(@"REFPROP.DLL")]
```
Comment (ianhbell) [inline wrappers/C#_High_Level_Wrapper/RefpropDLL/Refprop_DLL.cs:12]: This implies 32-bit, not what you want

**Review by ianhbell — CHANGES_REQUESTED** [review]

File: wrappers/C#_High_Level_Wrapper/RefpropDLL/Refprop_DLL.cs (hunk 7 above)
Comment (meyinyil) [reply wrappers/C#_High_Level_Wrapper/RefpropDLL/Refprop_DLL.cs:12]: Yes, I can do it for 64 bit too. If you require.

File: wrappers/C#_High_Level_Wrapper/Refprop_Example/Form1.cs (hunk 1 above)
Comment (meyinyil) [reply wrappers/C#_High_Level_Wrapper/Refprop_Example/Form1.cs]: It's been corrected. Thanks for the comment.

File: wrappers/C#_High_Level_Wrapper/LICENSE.txt (hunk 2 above)
Comment (meyinyil) [reply wrappers/C#_High_Level_Wrapper/LICENSE.txt:1]: Actually I can delete the license,

File: wrappers/C#_High_Level_Wrapper/README.md (hunk 4 above)
Comment (meyinyil) [reply wrappers/C#_High_Level_Wrapper/README.md]: The link is deleted. Thanks for the comment.

File: wrappers/C#_High_Level_Wrapper/README.md (hunk 5 above)
Comment (meyinyil) [reply wrappers/C#_High_Level_Wrapper/README.md]: (Bin/Debug)is deleted. Thanks for the comment.

**Comment by ianhbell** [comment]
Please remove again all the non-necessary files. The goal in this repo is to provide as little as possible, and a single MWE for each wrapper.

File: wrappers/C#_High_Level_Wrapper/Refprop_Example/Form1.cs (hunk 1 above)
Comment (ianhbell) [reply wrappers/C#_High_Level_Wrapper/Refprop_Example/Form1.cs]: Still a lot more left. Search the whole folder

File: wrappers/C#_High_Level_Wrapper/RefpropDLL/Refprop_DLL.cs (hunk 7 above)
Comment (ianhbell) [reply wrappers/C#_High_Level_Wrapper/RefpropDLL/Refprop_DLL.cs:12]: Yes, 32-bit is obsolete. Better to do only 64-bit if you are only going to do one.

**Comment by meyinyil** [comment]
Repo will be uploaded after requested changeds are made.

**Pull request closed without merging**